### PR TITLE
dmlive: init at unstable-2022-06-05

### PR DESCRIPTION
**Pull request by NickCao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/video/dmlive/default.nix [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  cargoHash = "sha256-Mkg/Ifr2mJRs97b+Dw6I1RuqXERpSigDEzGkr+wWN5w=";
+
+  OPENSSL_NO_VENDOR = true;
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
+  buildInputs = [ openssl ];
+
+  postInstall = ''
+    wrapProgram "$out/bin/dmlive" --prefix PATH : "${lib.makeBinPath [ mpv ffmpeg ]}"
```
Comment (jian-lin) [inline pkgs/applications/video/dmlive/default.nix:30]: `nodejs` is also needed, see https://github.com/THMonster/dmlive/blob/fd4fa1859f05350658db598a50d29f59d22b55a1/src/utils/mod.rs#L32

File: pkgs/applications/video/dmlive/default.nix [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  '';
+
+  meta = with lib; {
+    description = "Tools to play live stream with danmaku";
```
Comment (jian-lin) [inline pkgs/applications/video/dmlive/default.nix:34]:
```suggestion
    description = "A tool to play and record live streams with danmaku";
```

File: pkgs/applications/video/dmlive/default.nix [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+, pkg-config
+, makeWrapper
+, openssl
+, mpv
+, ffmpeg
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "dmlive";
+  version = "unstable-2022-06-05";
```
Comment (jian-lin) [inline pkgs/applications/video/dmlive/default.nix:13]: new commits are available

**Review by jian-lin — CHANGES_REQUESTED** [review]
LGTM

**Review by jian-lin — APPROVED** [review]

File: pkgs/applications/video/dmlive/default.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    rev = "fd4fa1859f05350658db598a50d29f59d22b55a1";
+    hash = "sha256-NVabHLxPHi7hWoztthPmVC5VRKQKglpytuUQOY1Hzrw=";
+  };
+
+  cargoHash = "sha256-TziP7n9Xgi/wHaiF/NI6noMp1iR6vRuAXxvKJwQHbTw=";
+
+  OPENSSL_NO_VENDOR = true;
+
+  nativeBuildInputs = [ pkg-config makeWrapper ];
+  buildInputs = [ openssl ];
```
Comment (SuperSandro2000) [inline pkgs/applications/video/dmlive/default.nix:28]: Can you try to fix the darwin build by adding ``Security`` framework for darwin only?

**Pull request merged**